File: app/src/main/java/forpdateam/ru/forpda/model/system/SchedulersProvider.java
package forpdateam.ru.forpda.model.system;

import io.reactivex.Scheduler;

/**
 * Created by radiationx on 01.01.18.
 */

public interface SchedulersProvider {
    Scheduler ui();

    Scheduler computation();

    Scheduler trampoline();

    Scheduler newThread();

    Scheduler io();
}

File: app/src/main/java/forpdateam/ru/forpda/model/SchedulersProvider.kt
package forpdateam.ru.forpda.model

import io.reactivex.Scheduler

/**
 * Created by radiationx on 01.01.18.
 */

interface SchedulersProvider {
    fun ui(): Scheduler
    fun computation(): Scheduler
    fun trampoline(): Scheduler
    fun newThread(): Scheduler
    fun io(): Scheduler
}
